pkg/operator/sync.go

```diff
@@ -612,8 +613,13 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig, _ *configv1.ClusterOpera
 		return err
 	}
 
+	moscs, err := optr.getAndValidateMachineOSConfigs()
```

Comment:
Hmm, does this need to be behind a FG?

Reply:
It may not look like it, but it technically is. The `optr.getAndValidateMachineOSConfigs()` function checks for the presence of the FeatureGate before trying to perform its query. If the FeatureGate is not enabled, it returns nil. I can add some comments or the like to indicate that is the case.